app/routes.py

```diff
@@ -0,0 +1,269 @@
+            return {"details": "Request body must include total_inventory."}, 400
+        new_video = Video(title=request_body["title"],
+                        release_date=request_body["release_date"],
+                        total_inventory=request_body["total_inventory"])
+        db.session.add(new_video)
+        db.session.commit()
+        return jsonify(new_video.to_json()), 201
+
+
+# GET VIDEO BY ID
+@videos_bp.route("/<id>", methods=["GET"])
+def get_video(id):
+    if id.isnumeric() != True:
+        return {"message": "Video id provided is not a number."}, 400
+    video = Video.query.get(id)
+    if video is None:
+        return {"message": f"Video {id} was not found"}, 404
+    if request.method == "GET":
+        return jsonify(video.to_json()), 200
+
+
+# UPDATE VIDEO BY ID
+@videos_bp.route("/<id>", methods=["PUT"])
+def update_video(id):
+    video = Video.query.get(id)
+    if video is None:
+        return {"message": f"Video {id} was not found"}, 404
+    if request.method == "PUT":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {"details": "Request body must include title."}, 400
+        elif "release_date" not in request_body:
+            return {"details": "Request body must include release_date."}, 400
+        elif "total_inventory" not in request_body:
+            return {"details": "Request body must include total_inventory."}, 400
+        video.title = request_body["title"]
+        video.request_body = request_body["release_date"]
+        video.total_inventory = request_body["total_inventory"]
+        db.session.commit()
+        return jsonify(video.to_json()), 200
+
+
+# DELETE VIDEO BY ID
+@videos_bp.route("/<id>", methods=["DELETE"])
+def delete_video(id):
+    video = Video.query.get(id)
+    if video is None:
+        return {"message": f"Video {id} was not found"}, 404
+    db.session.delete(video)
+    db.session.commit()
+    return {"id": video.id}, 200
+
+
+# --------------------------------
+# ----------- RENTALS ----------
+# --------------------------------
+
+
+# CHECK OUT A VIDEO TO A CUSTOMER
+@rentals_bp.route("/check-out", methods=["POST"])
+def checkout_video():
+    request_body = request.get_json()
+    if "customer_id" not in request_body:
+        return {"details": "Request body must include customer_id."}, 400
+    elif "video_id" not in request_body:
+        return {"details": "Request body must include video_id."}, 400
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+    # check if customer and video exist
+    if customer is None or video is None:
+        return make_response("", 404)
+    num_of_videos_rented = Rental.query.filter_by(
+        video_id=video.id, checked_out=True).count()
+    available_inventory = video.total_inventory - num_of_videos_rented
```

Comment:
Consider moving the logic for calculating the checked out count and available inventory to helper methods in the Rental model.